joomla/joomla-cms
# [4.0] Improving menu item workflow

New Pull Request for #11766

### Summary of Changes
This pull request adds a new button inside the EditArticle view to create a corresponding menu item directly.


### Testing Instructions

1. Install patch / PR
2. Go to _Content_ > _Articles_
3. Create a new article by clicking on _New_
![01 new article no button](https://user-images.githubusercontent.com/41572666/43576403-88b8a022-9649-11e8-8d07-da2c45da45f7.png)
As you can see, the button is not available due to no new article is created
4. Fill in the title and press _Save_
5. The editing screen of the newly created article shows up and you are now able to create a new menu item with the button in the toolbar
![02 edit article new menu item button](https://user-images.githubusercontent.com/41572666/43576457-bbb674fe-9649-11e8-841e-2b940e38a442.png)
6. If you click the button, the modal opens
![03 new menu item modal](https://user-images.githubusercontent.com/41572666/43577051-10e6eb4c-964b-11e8-996e-b5c353b9c06a.PNG)
7. Now set _Menu Title_, as _Menu Item Type_ choose _Single Article_ and select the new article
![04 set menu item title and article](https://user-images.githubusercontent.com/41572666/43577139-55c8b448-964b-11e8-8128-51578fcd65cb.PNG)
8. Click _Save & Close_ and head to _Menus_ > _[The Menu with the new Item]_.
You should see the new item in the list
![05 menu item created](https://user-images.githubusercontent.com/41572666/43577456-2c58c2be-964c-11e8-8c75-b1b66ddcf9e8.PNG)
9. If the menu is connected to a menu module you should see it on the site as shown below
![06 article and menu item published on site](https://user-images.githubusercontent.com/41572666/43577571-79c47e94-964c-11e8-9184-298c7833d69f.PNG)


Added benefit: you can create any type of menu link


### Expected result
The funcionality works as pointed out in the **Testing Instructions** section.


### Actual result
At the moment there is not such a functionality



### Documentation Changes Required
Yes, has been added to [docs.joomla.org](https://docs.joomla.org/Adding_a_menu_link_in_an_article)
